ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -57,19 +57,19 @@ export function useTransactionDisplayData (transactionGroup) {
   const t = useI18nContext()
   const { initialTransaction, primaryTransaction } = transactionGroup
   // initialTransaction contains the data we need to derive the primary purpose of this transaction group
-  const { transactionCategory } = initialTransaction
+  const transactionCategory = primaryTransaction?.transactionCategory || initialTransaction?.transactionCategory
```

Comment:
[Req] let's create a new variable above this that's just 'transaction' and set it to `primaryTransaction || initialTransaction` to reduce the complexity of the rest of the function.